### DISCUSSION TITLE
Question about the geometry in egs in micrometers

### DISCUSSION THREAD

**Discussion by ShahinGHD** [nrc-cnrc/EGSnrc#1052] [Questions]
Hello
I actually have three question:

1- Does egs 3ddose dose scoring work fine when I have voxels at the size of micrometers?
2- is it possible to have a big cubic phantom, and inside it, place some smaller cubes and both have voxels with the size of 1mm?
assume that the big box is 10x10x10 cm and inside it, i have three cubes with a size of 2x2x2 cm in each row (  three rows in total)
and all of them have 1mm voxel size for dose scoring. if so, how?

I tried making this with the Dosexyznrc GUI as an egsphant file but no luck.

3- if #2 is possible and I want to creat a TIA file to spread the activity only inside each small cube, how can I find the voxel numbers to creat that file? (voxel indices)

Thanks

**Comment by rtownson** [comment]
This is not possible with DOSXYZnrc, but it is with egs++. You can place an XYZ geometry with small voxel sizes inside a single voxel of another XYZ geometry using an envelope (just make sure that no surfaces are actually overlapping or perfectly aligned, use a tiny offset to avoid floating point errors). If you want to replace a row or section of voxels with smaller ones, use egs_gunion instead (but note that it's less computationally efficient). 

Voxel indices are assigned deterministically, so it is possible to understand the algorithm and know what voxel indices are of interest. Use egs_view to understand the numbering. Then use egs_dose_scoring to print the dose to all regions, or just to those regions, and use a script to parse them or re-order them for profiles etc. It's a bit tedious but very do-able.
